rustls/src/msgs/handshake.rs

```diff
@@ -284,6 +284,16 @@ impl ServerNamePayload<'_> {
     const SIZE_LEN: ListLength = ListLength::NonZeroU16 {
         empty_error: InvalidMessage::IllegalEmptyList("ServerNames"),
     };
+
+    /// Get the `DnsName` out of this `ServerNamePayload` if it contains one.
+    /// The returned `DnsName` will be normalized (converted to lowercase).
+    pub(crate) fn to_dns_name_normalized(&self) -> Option<DnsName<'static>> {
+        match self {
+            ServerNamePayload::SingleDnsName(dns_name) => Some(dns_name.to_lowercase_owned()),
```

Comment:
Not sure why this isn't triggering `clippy::use_self`...

Reply:
`Self` here refers to a different lifetime